GrassChoppers/AnalyticsTracker.swift

```diff
@@ -0,0 +1,100 @@
+import Foundation
+
+/// The set of supported analytics events.
+/// Each event has a mapping to its string representation.
+enum AnalyticsEvent: String {
+    
+    /// The app was launched.
+    case appLaunch = "app_launch"
+    
+    /// The app was sent to the background.
+    case appBackgrounded = "app_backgrounded"
+    
+    /// The app was launched from having been backgrounded.
+    case appResumed = "app_resumed"
+    
+    /// The user viewed the Home screen.
+    case homeScreenViewed = "home_screen_viewed"
+}
+
+/// The interface that anthing acting as an analytics tracker must have.
+protocol AnalyticsTracking {
+    
+    /// Records an analytics event.
+    /// - Parameter event: The event to be tracked.
+    func track(event: AnalyticsEvent)
+}
+
+/// A type for recording analytics events to the server.
+class AnalyticsTracker: AnalyticsTracking {
+    
+    /// The network session configuration ot be used when sending events.
+    private let urlSession: URLSession
```

Comment:
Why did we make the URLSession private again?

Reply:
`private` means that this variable is not accessible from outside of this class (ie. you cannot call tracker.urlSession on an instance of AnalyticsTracker).

This is because the usage of `AnalyticsTracker` does not require that you have access to that variable, and to simplify its usage and keep you from messing the tracker up, you cant even access that property from outside. The only code that really needs to use that `URLSession` is on the inside of the class.